src/mpi/pt2pt/sendrecv.c

```diff
@@ -173,3 +173,119 @@ int MPIR_Sendrecv_replace_impl(void *buf, int count, MPI_Datatype datatype, int
   fn_fail:
     goto fn_exit;
 }
+
+int MPIR_Isendrecv_impl(const void *sendbuf, int sendcount, MPI_Datatype sendtype,
+                        int dest, int sendtag,
+                        void *recvbuf, int recvcount, MPI_Datatype recvtype,
+                        int source, int recvtag, MPIR_Comm * comm_ptr, MPIR_Request ** p_req)
+{
+    int mpi_errno = MPI_SUCCESS;
+
+    if (unlikely(source == MPI_PROC_NULL)) {
+        /* recv from MPI_PROC_NULL, just send */
+        mpi_errno = MPID_Isend(sendbuf, sendcount, sendtype, dest, sendtag, comm_ptr,
+                               MPIR_CONTEXT_INTRA_PT2PT, p_req);
+        MPIR_ERR_CHECK(mpi_errno);
+        goto fn_exit;
+    } else if (unlikely(dest == MPI_PROC_NULL)) {
+        /* send to MPI_PROC_NULL, just recv */
+        mpi_errno = MPID_Irecv(recvbuf, recvcount, recvtype, source, recvtag, comm_ptr,
+                               MPIR_CONTEXT_INTRA_PT2PT, p_req);
+        MPIR_ERR_CHECK(mpi_errno);
+        goto fn_exit;
+    }
+
+    MPIR_Sched_t s = MPIR_SCHED_NULL;
+    int sched_tag;
+
+    mpi_errno = MPIR_Sched_next_tag(comm_ptr, &sched_tag);
```

Comment:
I don't think `sched_tag` has any relevance to these functions. Can we avoid allocating one?

Reply:
Probably. I blindly copied it without too much thinking :)

Reply:
`sched_tag` removed.